### PR TITLE
Previous winners

### PR DESCRIPTION
**Pull request by jaleszek**
Company details form.

TODO: clarify assignment the form to specific year with Michael

### CODE REVIEW

File: app/assets/javascripts/admin/form_answers.js.coffee [hunk 1]
```diff
@@ -175,4 +181,48 @@ bindRags =(klass) ->
         $(this).parents("select").val(ragClicked)
     $(klass).submit()
 
+
+handleCompanyDetailsForm =->
```
Comment (matthewford) [inline app/assets/javascripts/admin/form_answers.js.coffee]: Function arrow (->) must be spaced properly

File: app/controllers/concerns/company_details_mixin.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module CompanyDetailsMixin
+  def update
```
Comment (matthewford) [inline app/controllers/concerns/company_details_mixin.rb]: Assignment Branch Condition size for update is too high. [16.16/15]

File: app/controllers/concerns/form_answer_mixin.rb [hunk 3]
```diff
@@ -13,13 +12,23 @@ def update
         }
       end
 
-      format.html { redirect_to [namespace_name, resource] }
+      format.html do
+        if request.xhr?
+          render partial: "admin/form_answers/company_details/#{params[:section]}_form", layout: false
```
Comment (matthewford) [inline app/controllers/concerns/form_answer_mixin.rb]: Line is too long. [102/100]

File: app/models/form_answer.rb [hunk 4]
```diff
@@ -119,6 +120,12 @@ def state_machine
     end
   end
 
+  accepts_nested_attributes_for :previous_wins,
+                                allow_destroy: true,
+                                reject_if: proc { |attrs|
```
Comment (matthewford) [inline app/models/form_answer.rb]: Avoid using {...} for multi-line blocks.

File: app/models/form_answer.rb [hunk 5]
```diff
@@ -48,6 +48,7 @@ class FormAnswer < ActiveRecord::Base
     has_one :feedback, dependent: :destroy
     has_one :press_summary, dependent: :destroy
     has_one :draft_note, as: :notable, dependent: :destroy
+    has_one :company_detail, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 6]
```diff
@@ -57,7 +58,7 @@ class FormAnswer < ActiveRecord::Base
     has_many :comments, as: :commentable, dependent: :destroy
     has_many :form_answer_transitions
     has_many :assessor_assignments, dependent: :destroy
-
+    has_many :previous_wins, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

**Comment by jaleszek** [comment]
@RuslanHamidullin take a look please when you get a chance. Thanks

**Comment by RuslanHamidullin** [comment]
@jaleszek sure

**Comment by RuslanHamidullin** [comment]
Looks good :+1:

**Pull request merged**